### PR TITLE
Expose local socket.

### PR DESCRIPTION
**Pull request by cmeiklejohn**

### CODE REVIEW

**Comment by cmeiklejohn** [comment]
@Vagabond updated pull request to return just the IP, which matches `peer/1`, and store to the `sockname` in the record once a lookup has been performed.

File: src/webmachine_request.erl [hunk 1]
```diff
@@ -130,6 +130,23 @@ get_peer({?MODULE, ReqState}=Req) ->
 get_peer(ReqState) ->
     get_peer({?MODULE, ReqState}).
 
+get_sock({?MODULE, ReqState} = Req) ->
+    case ReqState#wm_reqstate.sock of
```
Comment (reiddraper) [inline src/webmachine_request.erl:134]: weird indentation in this case statement

Comment (cmeiklejohn) [inline src/webmachine_request.erl:134]: Resolved in [947b540].

File: src/webmachine_request.erl [hunk 2]
```diff
@@ -130,6 +130,23 @@ get_peer({?MODULE, ReqState}=Req) ->
...
+        undefined ->
+            Sockname = case ReqState#wm_reqstate.socket of
+                testing -> {ok, {{127,0,0,1}, 80}};
+                {ssl,SslSocket} -> ssl:sockname(SslSocket);
```
Comment (Vagabond) [inline src/webmachine_request.erl:138]: This is marginally naughty, you should not peek at the internals of the SSL socket type in case they change it.

Comment (cmeiklejohn) [inline src/webmachine_request.erl:138]: Agreed, however this is currently how `get_peer` is handled as well.

Comment (cmeiklejohn) [inline src/webmachine_request.erl:138]: It was more a stylistic choice to follow existing conventions.

**Comment by cmeiklejohn** [comment]
Is there anything I can do to help this one along? @reiddraper @kellymclaughlin

**Comment by reiddraper** [comment]
+1 from me

**Comment by cmeiklejohn** [comment]
Nice.  What's the best way to get this into Riak proper?  Do I need to merge against another branch?

**Comment by reiddraper** [comment]
> Nice. What's the best way to get this into Riak proper? Do I need to merge against another branch?

It looks like we've just been using tags, like [here](https://github.com/basho/riak_core/blob/master/rebar.config#L14-L15). So I'd think we'd just create another tag from master before 2.0.

**Comment by cmeiklejohn** [comment]
Cool; I'm going to merge this.

**Pull request merged**